### PR TITLE
Don't mistake a following if for an elif

### PR DESCRIPTION
**Pull request by konstin**
In the following code, the comment used to get wrongly associated with the `if False` since it looked like an elif. This fixes it by checking the indentation and adding a regression test
```python
if True:
    pass
else:  # Comment
    if False:
        pass
    pass
```
    
Originally found in https://github.com/gradio-app/gradio/blob/1570b94a02d23d051ae137e0063974fd8a48b34e/gradio/external.py#L478

### CODE REVIEW

**Comment by konstin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5293** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/5293" target="_blank">Graphite</a> 
    * **PR #5296** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/5296" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/5296?utm_source=stack-comment).

File: crates/ruff_python_formatter/resources/test/fixtures/ruff/expression/dict.py [hunk 1]
```diff
@@ -48,3 +48,11 @@
     C: 0.1 * (10.0 / 12),
     D: 0.1 * (10.0 / 12),
 }
+
+# Regression test for formatter panic with comment after parenthesized dict value
+# Originally found in https://github.com/bolucat/Firefox/blob/636a717ef025c16434997dc89e42351ef740ee6b/testing/marionette/client/marionette_driver/geckoinstance.py#L109
+a = {
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/resources/test/fixtures/ruff/expression/dict.py:54]: This fix seems unrelated to if/elif right?

File: crates/ruff_python_formatter/src/comments/placement.rs [hunk 2]
```diff
@@ -335,38 +335,37 @@ fn handle_in_between_bodies_end_of_line_comment<'a>(
...
+                //     pass
+                // else:  # Comment
+                //     if False:
+                //         pass
                 //     pass
                 // ```
-                CommentPlacement::Default(comment)
-            } else {
-                CommentPlacement::trailing(preceding, comment)
+                let base_if_indent =
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/src/comments/placement.rs:359]: Just an idea. It may be annoying to do because of all the type narrowing that's necessary and you would need to identify the right parent body but I thought it may be worth mentioning. The way to normally detect whether it is an `elif` or not... oh no.... It is not. These two programs have the same AST

```python
if True:
    pass
elif False:
    pass
```

```
If(
    StmtIf {
        range: 0..38,
        test: Constant(
            ExprConstant {
                range: 3..7,
                value: Bool(
                    true,
                ),
                kind: None,
            },
        ),
        body: [
            Pass(
                StmtPass {
                    range: 13..17,
                },
            ),
        ],
        orelse: [
            If(
                StmtIf {
                    range: 18..38,
                    test: Constant(
                        ExprConstant {
                            range: 23..28,
                            value: Bool(
                                false,
                            ),
                            kind: None,
                        },
                    ),
                    body: [
                        Pass(
                            StmtPass {
                                range: 34..38,
                            },
                        ),
                    ],
                    orelse: [],
                },
            ),
        ],
    },
),
```

```python
if True:
    pass
else:
    if False:
        pass
```

```
If(
        StmtIf {
            range: 41..91,
            test: Constant(
                ExprConstant {
                    range: 44..48,
                    value: Bool(
                        true,
                    ),
                    kind: None,
                },
            ),
            body: [
                Pass(
                    StmtPass {
                        range: 54..58,
                    },
                ),
            ],
            orelse: [
                If(
                    StmtIf {
                        range: 69..91,
                        test: Constant(
                            ExprConstant {
                                range: 72..77,
                                value: Bool(
                                    false,
                                ),
                                kind: None,
                            },
                        ),
                        body: [
                            Pass(
                                StmtPass {
                                    range: 87..91,
                                },
                            ),
                        ],
                        orelse: [],
                    },
                ),
            ],
        },
    ),
```

**Review by MichaReiser — APPROVED** [review]

File: crates/ruff_python_formatter/resources/test/fixtures/ruff/expression/dict.py (hunk 1 above)
Comment (konstin) [reply crates/ruff_python_formatter/resources/test/fixtures/ruff/expression/dict.py:54]: it's already merged: https://github.com/astral-sh/ruff/pull/5293/files#diff-eb676c05c3588ddfa4f15b679292a4418e3b3d1bc5e17ce2ad9e150ff128097aR53

Comment (konstin) [reply crates/ruff_python_formatter/resources/test/fixtures/ruff/expression/dict.py:54]: just confirmed, it's also not part of the merge commit
